node/malus/src/variants/suggest_garbage_candidate.rs

```diff
@@ -88,131 +92,166 @@ where
-					?relay_parent,
-					"Received request to second candidate"
+					"😈 Started Malus node with '--percentage' set to: {:?}",
+					&self.percentage,
 				);
 
-				let pov = PoV { block_data: BlockData(MALICIOUS_POV.into()) };
-
-				let (sender, receiver) = std::sync::mpsc::channel();
-				let mut new_sender = subsystem_sender.clone();
-				let _candidate = candidate.clone();
-				self.spawner.spawn_blocking(
-					"malus-get-validation-data",
-					Some("malus"),
-					Box::pin(async move {
-						gum::trace!(target: MALUS, "Requesting validators");
-						let n_validators = request_validators(relay_parent, &mut new_sender)
-							.await
-							.await
-							.unwrap()
-							.unwrap()
-							.len();
-						gum::trace!(target: MALUS, "Validators {}", n_validators);
-						match find_validation_data(&mut new_sender, &_candidate.descriptor()).await
-						{
-							Ok(Some((validation_data, validation_code))) => {
-								sender
-									.send((validation_data, validation_code, n_validators))
-									.expect("channel is still open");
-							},
-							_ => {
-								panic!("Unable to fetch validation data");
-							},
-						}
-					}),
-				);
-
-				let (validation_data, validation_code, n_validators) = receiver.recv().unwrap();
-
-				let validation_data_hash = validation_data.hash();
-				let validation_code_hash = validation_code.hash();
-				let validation_data_relay_parent_number = validation_data.relay_parent_number;
-
-				gum::trace!(
+				gum::debug!(
 					target: MALUS,
 					candidate_hash = ?candidate.hash(),
 					?relay_parent,
-					?n_validators,
-					?validation_data_hash,
-					?validation_code_hash,
-					?validation_data_relay_parent_number,
-					"Fetched validation data."
+					"Received request to second candidate",
 				);
 
-				let malicious_available_data =
-					AvailableData { pov: Arc::new(pov.clone()), validation_data };
-
-				let pov_hash = pov.hash();
-				let erasure_root = {
-					let chunks =
-						erasure::obtain_chunks_v1(n_validators as usize, &malicious_available_data)
-							.unwrap();
-
-					let branches = erasure::branches(chunks.as_ref());
-					branches.root()
-				};
-
-				let (collator_id, collator_signature) = {
-					use polkadot_primitives::v2::CollatorPair;
-					use sp_core::crypto::Pair;
-
-					let collator_pair = CollatorPair::generate().0;
-					let signature_payload = polkadot_primitives::v2::collator_signature_payload(
-						&relay_parent,
-						&candidate.descriptor().para_id,
-						&validation_data_hash,
-						&pov_hash,
-						&validation_code_hash,
-					);
+				// Need to draw value from Bernoulli distribution with given probability of success defined by the Clap parameter.
+				// Note that clap parameter must be f64 since this is expected by the Bernoulli::new() function, hence it must be converted.
+				let distribution = Bernoulli::new(self.percentage / 100.0).expect("Invalid probability! Percentage cannot be < 0 or > 100.");
```

Comment:
Why do we need a bernoulli distribution in the first place? A simple thing is to convert percents into a range [0..1] and then generate a random float in a range [0..1] and compare that float with the sample probability (like `rng_number < sample_probability`). Since our PRNG is uniform, we will get roughly the same distribution at no additional cost.

Reply:
Indeed, then the approach with the actual distribution is more expressive and is really preferred.